src/fidasim.f90

```diff
@@ -2301,6 +2385,19 @@ subroutine read_equilibrium
     call h5ltread_dataset_double_f(gid, "/fields/ez", equil%fields%ez, dims, error)
     call h5ltread_dataset_int_f(gid, "/fields/mask", f_mask, dims,error)
 
+    !!Calculate B field derivatives
+    do iz=1,inter_grid%nz
+        call deriv(inter_grid%r, equil%fields(:,iz)%br, equil%fields(:,iz)%dbr_dr)
+        call deriv(inter_grid%r, equil%fields(:,iz)%bt, equil%fields(:,iz)%dbt_dr)
+        call deriv(inter_grid%r, equil%fields(:,iz)%bz, equil%fields(:,iz)%dbz_dr)
+    enddo
+
+    do ir=1,inter_grid%nr
+        call deriv(inter_grid%z, equil%fields(ir,:)%br, equil%fields(ir,:)%dbr_dz)
+        call deriv(inter_grid%z, equil%fields(ir,:)%bt, equil%fields(ir,:)%dbt_dz)
+        call deriv(inter_grid%z, equil%fields(ir,:)%bz, equil%fields(ir,:)%dbz_dz)
+    enddo
+
```

Comment:
You have a routine to calculate derivatives in two dimensions why not use those?